include/unifex/let_with_stop_source.hpp

```diff
@@ -129,7 +129,7 @@ class _stop_source_sender<SuccessorFactory>::type {
     }
 
 private:
-    UNIFEX_NO_UNIQUE_ADDRESS SuccessorFactory func_;
+    UNIFEX_NO_UNIQUE_ADDRESS remove_cvref_t<SuccessorFactory> func_;
```

Comment:
Is it correct to strip qualifiers here but not on line 108 or lines 124-127? It's probably better to strip qualifiers once in the definition of `stop_source_sender`:

```c++
template<typename SuccessorFactory>
using stop_source_sender =
    typename _stop_source_sender<remove_cvref_t<SuccessorFactory>>::type;
```

Reply:
Lewis has been tending to strip them as late as possible. 

You may be right that what's here is not correct, though - it is what I realised was missing for correctness to fix a bug.

Reply:
I think it is right here to store the func as a value, but to keep the cvrefness in the templates otherwise. I've updated it on that basis.

Reply:
I have been tending to have the qualifiers stripped by the callsite that is instantiating the sender/receiver class rather than implicitly inside aliases or when defining data-members.

One of the reasons is to try to avoid instantiating senders/receivers multiple times when constructed with different value-categories. eg. if we are stripping qualifiers at the data-member level then invoking `let_with_stop_source(f)` and `let_with_stop_source(std::move(f))` will return two different sender-types even though all of the logic is identical except for the sender constructor which moves rather than copies.

But if we always return `_stop_source_sender<decay_t<SuccessorFactory>>` from `_fn::operator()` then you'll just get the one instantiation of the class and only instantiate the constructor multiple times.